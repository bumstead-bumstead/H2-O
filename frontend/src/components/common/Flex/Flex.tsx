import type { HTMLAttributes } from 'react';
import styled from '@emotion/styled';
import type { BoxStyle, BoxStyleProperties } from '@/types/boxStyle';
import type { FlexStyle, FlexStyleProperties } from '@/types/flexStyle';
import type { PositionStyle, PositionStyleProperties } from '@/types/positionStyle';

type Props = FlexStyle & BoxStyle & PositionStyle & HTMLAttributes<HTMLDivElement>;

const componentStyle = ({
  flexDirection,
  justifyContent,
  alignContent,
  alignItems,
  flexWrap,
  gap,
  position,
  top,
  bottom,
  left,
  right,
  backgroundColor,
  width,
  height,
  marginBottom,
  marginLeft,
  marginRight,
  marginTop,
  paddingBottom,
  paddingLeft,
  paddingRight,
  paddingTop,
}: Pick<Props, FlexStyleProperties | BoxStyleProperties | PositionStyleProperties>) => {
  return {
    display: 'flex',
    flexDirection,
    justifyContent,
    alignContent,
    alignItems,
    flexWrap,
    gap,
    position,
    top,
    bottom,
    left,
    right,
    backgroundColor,
    width,
    height,
    marginBottom,
    marginLeft,
    marginRight,
    marginTop,
    paddingBottom,
    paddingLeft,
    paddingRight,
    paddingTop,
  };
};

const Flex = styled.div<Props>((props) => {
  return componentStyle(props);
});

export default Flex;
