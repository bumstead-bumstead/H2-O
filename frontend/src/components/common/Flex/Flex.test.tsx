import Flex from './Flex';
import { render, screen } from '@/tests/test-util';

describe('Flex 컴포넌트 테스트', () => {
  it('Flex가 정상적으로 렌더링 된다.', () => {
    render(<Flex>Flex</Flex>);

    expect(screen.getByText('Flex')).toBeInTheDocument();
  });

  it('Flex가 display flex속성을 가지고 있다.', () => {
    render(<Flex>Flex</Flex>);

    expect(screen.getByText('Flex')).toHaveStyleRule('display', 'flex');
  });

  it('Flex 스냅샷 테스트', () => {
    render(<Flex>Flex</Flex>);

    expect(screen.getByText('Flex')).toMatchSnapshot();
  });
});
