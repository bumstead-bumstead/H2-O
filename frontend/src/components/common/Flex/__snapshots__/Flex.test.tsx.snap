// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Flex 컴포넌트 테스트 > Flex 스냅샷 테스트 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<div
  class="emotion-0"
>
  Flex
</div>
`;
